src/controls/MenuSelectHeading.tsx

```diff
@@ -173,19 +174,21 @@ export default function MenuSelectHeading({
     // show the first of the selected nodes' levels and not allow changing all
     // selected to that heading level. See
     // https://github.com/ueberdosis/tiptap/issues/3481.)
-    const level = numCurrentNodeLevels === 1 ? currentNodeLevels[0] : undefined;
-    if (level && level in LEVEL_TO_HEADING_OPTION_VALUE) {
+    currentLevel =
+      numCurrentNodeLevels === 1 ? currentNodeLevels[0] : undefined;
+    if (currentLevel && currentLevel in LEVEL_TO_HEADING_OPTION_VALUE) {
       selectedValue =
         LEVEL_TO_HEADING_OPTION_VALUE[
-          level as keyof typeof LEVEL_TO_HEADING_OPTION_VALUE
+          currentLevel as keyof typeof LEVEL_TO_HEADING_OPTION_VALUE
         ];
     }
   }
 
   const isCurrentlyParagraphOrHeading = selectedValue !== "";
   const canSetParagraph = editor?.can().setParagraph();
   // We have to pass a level when running `can`, so this is just an arbitrary one
-  const canSetHeading = editor?.can().setHeading({ level: 1 });
+  const canSetHeading =
```

Comment:
Would you be able to add a comment about why we're using this logic and link to the GitHub issue?

Reply:
OK! I'll work on it.

Reply:
I added a comment. Please check.